contracts/contracts/strategies/BridgedWOETHStrategy.sol

```diff
@@ -0,0 +1,303 @@
+        require(oraclePrice > 1 ether, "Invalid wOETH value");
+
+        uint128 oraclePrice128 = oraclePrice.toUint128();
+
+        // Do some checks
+        if (lastOraclePrice > 0) {
+            // Make sure the value only goes up
+            require(lastOraclePrice <= oraclePrice128, "Negative wOETH yield");
+
+            // And that it's within the bounds.
+            require(
+                // oraclePrice <= (lastOraclePrice * (1 + maxPriceDiffBps))
+                oraclePrice128 <=
+                    ((lastOraclePrice * (1e4 + uint128(maxPriceDiffBps))) /
+                        1e4),
+                "Price diff beyond threshold"
+            );
+        }
+
+        emit WOETHPriceUpdated(lastOraclePrice, oraclePrice128);
+
+        // Store the price
+        lastOraclePrice = oraclePrice128;
+
+        return oraclePrice;
+    }
+
+    /**
+     * @dev Wrapper for _updateWOETHOraclePrice with nonReentrant flag
+     * @return The latest price of wOETH from Oracle
+     */
+    function updateWOETHOraclePrice() external nonReentrant returns (uint256) {
+        return _updateWOETHOraclePrice();
+    }
+
+    /**
+     * @dev Computes & returns the value of given wOETH in WETH
+     * @param woethAmount Amount of wOETH
+     * @return Value of wOETH in WETH (using the last stored oracle price)
+     */
+    function getBridgedWOETHValue(uint256 woethAmount)
+        public
+        view
+        returns (uint256)
+    {
+        return (woethAmount * lastOraclePrice) / 1 ether;
+    }
+
+    /**
+     * @dev Takes in bridged wOETH and mints & returns
+     *      equivalent amount of OETHb.
+     * @param woethAmount Amount of bridged wOETH to transfer in
+     */
+    function depositBridgedWOETH(uint256 woethAmount)
+        external
+        onlyGovernorOrStrategist
+        nonReentrant
+    {
+        // Update wOETH price
+        uint256 oraclePrice = _updateWOETHOraclePrice();
+
+        // Figure out how much they are worth
+        uint256 oethToMint = (woethAmount * oraclePrice) / 1 ether;
+
+        // There's no pToken, however, it just uses WOETH address in the event
+        emit Deposit(address(weth), address(bridgedWOETH), oethToMint);
+
+        // Mint OETHb tokens and transfer it to the caller
+        IVault(vaultAddress).mintForStrategy(oethToMint);
+
+        // Transfer out minted OETHb
+        oethb.transfer(msg.sender, oethToMint);
+
+        // Transfer in all bridged wOETH tokens
+        bridgedWOETH.transferFrom(msg.sender, address(this), woethAmount);
+    }
+
+    /**
+     * @dev Takes in OETHb and burns it and returns
+     *      equivalent amount of bridged wOETH.
+     * @param woethAmount Amount of bridged wOETH to get back
+     */
+    function withdrawBridgedWOETH(uint256 woethAmount)
+        external
+        onlyGovernorOrStrategist
+        nonReentrant
+    {
+        // Update wOETH price
+        uint256 oraclePrice = _updateWOETHOraclePrice();
```

Comment:
👍 This is very cool that it intentionally breaks the funds moving in and out of it if the oracles are doing something weird. Very good.